src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ManagedIndexMetaData.kt

```diff
@@ -229,12 +254,23 @@ data class ManagedIndexMetaData(
                     POLICY_PRIMARY_TERM -> policyPrimaryTerm = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.longValue()
                     POLICY_COMPLETED -> policyCompleted = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.booleanValue()
                     ROLLED_OVER -> rolledOver = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.booleanValue()
-                    TRANSITION_TO -> transitionTo = xcp.text()
-                    StateMetaData.STATE -> state = StateMetaData.parse(xcp)
-                    ActionMetaData.ACTION -> action = ActionMetaData.parse(xcp)
-                    StepMetaData.STEP -> step = StepMetaData.parse(xcp)
-                    PolicyRetryInfoMetaData.RETRY_INFO -> retryInfo = PolicyRetryInfoMetaData.parse(xcp)
+                    TRANSITION_TO -> transitionTo = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.text()
+                    StateMetaData.STATE -> {
+                        // check null for invalid policy situation
+                        state = if (xcp.currentToken() == Token.VALUE_NULL) null else StateMetaData.parse(xcp)
+                    }
+                    ActionMetaData.ACTION -> {
+                        action = if (xcp.currentToken() == Token.VALUE_NULL) null else ActionMetaData.parse(xcp)
+                    }
+                    StepMetaData.STEP -> {
+                        step = if (xcp.currentToken() == Token.VALUE_NULL) null else StepMetaData.parse(xcp)
+                    }
+                    PolicyRetryInfoMetaData.RETRY_INFO -> {
+                        retryInfo = PolicyRetryInfoMetaData.parse(xcp)
+                    }
                     INFO -> info = xcp.map()
+                    // below line will break when getting metadata for explain or history
```

Comment:
notice that we also don't have this previously. Mainly because of some tests, it call expalin API and parse the response json, but it contained field we won't parse, for example the policyID `index.opendistro.index_state_management.policy_id`

Reply:
When do we parse metadata with index.opendistro.index_state_management.policy_id?
If I remember I thought that was added on to the explain response separately?

Reply:
in the test haha, we call explain API and parse it